### PR TITLE
Weak cipher detection in javax.crypto.KeyGenerator

### PR DESCRIPTION
**Pull request by DDJavierSantos**
# What Does This Do
Instruments javax.crypto.KeyGenerator so we can detect weak cypher vulnerabilities

# Motivation
We need to cover all of the calls where the user can create a cypher with a weak algorithm including KeyGenerator and SecretKeyFactory

# Additional Notes

### CODE REVIEW

File: dd-java-agent/instrumentation/java-security/src/main/java/datadog/trace/instrumentation/java/security/WeakCipherInstrumentationCallSite.java [hunk 1]
```diff
@@ -31,6 +31,46 @@ public static void beforeGetInstance(
     onCipherAlgorithm(algo);
   }
 
+  @CallSite.Before(
```
Comment (manuel-alvarez-alvarez) [inline dd-java-agent/instrumentation/java-security/src/main/java/datadog/trace/instrumentation/java/security/WeakCipherInstrumentationCallSite.java:34]: Maybe you can add some smoke tests for this

**Review by manuel-alvarez-alvarez — APPROVED** [review]

**Review by smola — APPROVED** [review]

File: dd-java-agent/instrumentation/java-security/src/main/java/datadog/trace/instrumentation/java/security/WeakCipherInstrumentationCallSite.java (hunk 1 above)
Comment (DDJavierSantos) [reply dd-java-agent/instrumentation/java-security/src/main/java/datadog/trace/instrumentation/java/security/WeakCipherInstrumentationCallSite.java:34]: done

File: dd-java-agent/instrumentation/java-security/src/main/java/datadog/trace/instrumentation/java/security/WeakCipherInstrumentationCallSite.java [hunk 2]
```diff
@@ -31,6 +31,46 @@ public static void beforeGetInstance(
...
+      "javax.crypto.KeyGenerator  javax.crypto.KeyGenerator.getInstance(java.lang.String)")
+  public static void beforeKeyGeneratorGetInstance(@CallSite.Argument final String algo) {
+    onCipherAlgorithm(algo);
+  }
+
+  @CallSite.Before(
```
Comment (manuel-alvarez-alvarez) [inline dd-java-agent/instrumentation/java-security/src/main/java/datadog/trace/instrumentation/java/security/WeakCipherInstrumentationCallSite.java:40]: If you are using `@Before` call sites, you can merge all this call sites in one:

```java
  @CallSite.Before("javax.crypto.KeyGenerator javax.crypto.KeyGenerator.getInstance(java.lang.String)")
  @CallSite.Before("javax.crypto.KeyGenerator javax.crypto.KeyGenerator.getInstance(java.lang.String, java.lang.String)")
  @CallSite.Before("javax.crypto.KeyGenerator javax.crypto.KeyGenerator.getInstance(java.lang.String, java.security.Provider)")
  ...
  public static void beforeKeyGeneratorGetInstance(@CallSite.Argument(0) final String algo) {
    onCipherAlgorithm(algo);
  }
```

**Review by manuel-alvarez-alvarez — APPROVED** [review]

File: dd-java-agent/instrumentation/java-security/src/main/java/datadog/trace/instrumentation/java/security/WeakCipherInstrumentationCallSite.java (hunk 2 above)
Comment (DDJavierSantos) [reply dd-java-agent/instrumentation/java-security/src/main/java/datadog/trace/instrumentation/java/security/WeakCipherInstrumentationCallSite.java:21]: suggestion applied

**Pull request merged**